browser/script_injector/script_injector_browsertest.cc

```diff
@@ -0,0 +1,163 @@
+  return script_injector_remote;
+}
+
+std::unique_ptr<net::test_server::HttpResponse> HandleRequest(
+    const net::test_server::HttpRequest& request) {
+  auto response = std::make_unique<net::test_server::BasicHttpResponse>();
+  response->set_code(net::HTTP_OK);
+  response->set_content(R"(
+                          <html>
+                           <head><title>OK</title></head>
+                          </html>
+                        )");
+  response->set_content_type("text/html; charset=utf-8");
+  return response;
+}
+
+class ScriptInjectorBrowserTest : public PlatformBrowserTest {
+ public:
+  ScriptInjectorBrowserTest()
+      : https_server_(net::EmbeddedTestServer::TYPE_HTTPS) {}
+
+  void SetUpOnMainThread() override {
+    PlatformBrowserTest::SetUpOnMainThread();
+
+    mock_cert_verifier_.mock_cert_verifier()->set_default_result(net::OK);
+    host_resolver()->AddRule("*", "127.0.0.1");
+    https_server_.RegisterRequestHandler(base::BindRepeating(HandleRequest));
+    content::SetBrowserClientForTesting(&client_);
+
+    ASSERT_TRUE(https_server_.Start());
+  }
+
+  void SetUpCommandLine(base::CommandLine* command_line) override {
+    PlatformBrowserTest::SetUpCommandLine(command_line);
+    mock_cert_verifier_.SetUpCommandLine(command_line);
+  }
+
+  void SetUpInProcessBrowserTestFixture() override {
+    PlatformBrowserTest::SetUpInProcessBrowserTestFixture();
+    mock_cert_verifier_.SetUpInProcessBrowserTestFixture();
+  }
+
+  void TearDownInProcessBrowserTestFixture() override {
+    mock_cert_verifier_.TearDownInProcessBrowserTestFixture();
+    PlatformBrowserTest::TearDownInProcessBrowserTestFixture();
+  }
+
+  content::WebContents* web_contents() {
+    return chrome_test_utils::GetActiveWebContents(this);
+  }
+
+ protected:
+  net::EmbeddedTestServer https_server_;
+
+ private:
+  BraveContentBrowserClient client_;
+  content::ContentMockCertVerifier mock_cert_verifier_;
+};
+
+// TESTS
+
+IN_PROC_BROWSER_TEST_F(ScriptInjectorBrowserTest, ScriptInjected) {
+  base::RunLoop run_loop;
+  auto callback = base::BindLambdaForTesting([&](base::Value value) {
+    EXPECT_TRUE(value.GetBool());
+    run_loop.Quit();
+  });
+  auto script = base::StringPrintf(kScript, "true");
+  const GURL url = https_server_.GetURL("a.com", "/");
+  ASSERT_TRUE(content::NavigateToURL(web_contents(), url));
+  auto remote = GetRemote(web_contents()->GetPrimaryMainFrame());
+  remote->RequestAsyncExecuteScript(
+      ISOLATED_WORLD_ID_BRAVE_INTERNAL, base::UTF8ToUTF16(std::string(script)),
+      blink::mojom::UserActivationOption::kDoNotActivate,
+      blink::mojom::PromiseResultOption::kAwait, std::move(callback));
+  // Wait until the callback is called.
+  run_loop.Run();
+  std::u16string expected_title(u"test");
+  content::TitleWatcher watcher(web_contents(), expected_title);
+  EXPECT_EQ(expected_title, watcher.WaitAndGetTitle());
+}
+
+IN_PROC_BROWSER_TEST_F(ScriptInjectorBrowserTest, ScriptInjectedReturnsFalse) {
```

Comment:
I'm not sure this test adds a lot of value. You just happen to have chosen a bool in the previous test so I think the correct result is sufficient. I think it would be more useful to test passing an empty callback? I think you could use `base::MockCallback` and mock `is_null` to return `true` and then check to make sure it never gets called?

Reply:
Added a new test for a different return type (dictionary), and unit tests for the `CheckIfWantResult` method.